app/cdap/components/shared/ConfirmDialog/index.tsx

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { ReactNode, useState } from 'react';
+import isObject from 'lodash/isObject';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogActions from '@material-ui/core/DialogActions';
+import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
+import KeyboardArrowUpIcon from '@material-ui/icons/KeyboardArrowUp';
+
+import IconButton from '@material-ui/core/IconButton';
+import { StyledBox, StyledDialog, StyledAlert } from './styles';
+import PrimaryTextButton from 'components/shared/Buttons/PrimaryTextButton';
+
+interface IConfirmDialogProps {
+  headerTitle: string | ReactNode;
+  isOpen: boolean;
+  cancelButtonText: string | ReactNode;
+  cancelFn: (arg0: any) => void;
+  confirmButtonText: string | ReactNode;
+  confirmFn: (arg0: any) => void;
+  confirmationElem?: any;
+  confirmationText?: string | ReactNode;
+  errorMessage?: string | ReactNode;
+  extendedMessage?: { response: string };
+  disableAction?: boolean;
+}
+
+export const ConfirmDialog = (Props: IConfirmDialogProps) => {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  const handleToggleExtendedMessage = () => {
+    setIsExpanded(!isExpanded);
+  };
+
+  const getExtendedMessage = () => {
+    if (Props.extendedMessage) {
+      return (
+        <>
+          <IconButton
+            size="small"
+            onClick={handleToggleExtendedMessage}
+            color="inherit"
+            title={isExpanded ? 'Hide details' : 'Show details'}
+          >
+            {isExpanded ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
+          </IconButton>
+          {isExpanded && (
+            <StyledBox>
+              {isObject(Props.extendedMessage) ? (
+                <pre>{Props.extendedMessage.response}</pre>
+              ) : (
+                <pre>{Props.extendedMessage}</pre>
+              )}
+            </StyledBox>
+          )}
+        </>
+      );
+    }
+  };
+
+  return (
+    <StyledDialog open={Props.isOpen} fullWidth>
+      <DialogTitle>{Props.headerTitle}</DialogTitle>
+      {Props.errorMessage && (
+        <StyledAlert severity="error">
```

Comment:
Make this configurable? I.e. expose `severity` as a prop and call `errorMessage` `statusMessage` or similar.